.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+default_install_hook_types: ["pre-push"]
+exclude: '(^vendor|.*golden$)'
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.2.0
+  hooks:
+  - id: check-added-large-files
+  - id: check-yaml
+    args: ["--allow-multiple-documents"]
+    # Ignore yaml files with {{ }} cause they're not valid.
+    exclude: '(test\/gohelloworld\/gohelloworld-chart\/templates\/.*yaml$)'
+  - id: check-shebang-scripts-are-executable
```

Comment:
I'm curious how this hook know which scripts need to be checked? Does it scan all the files' content?

Reply:
No, AFAIK it perform [checks](https://github.com/pre-commit/pre-commit-hooks/blob/main/pre_commit_hooks/check_executables_have_shebangs.py) only against currently staged files
and only if file starts with shebang with respect of [`git config fileMode`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-corefileMode)